src/opencensus/proto/error/v1/error.proto

```diff
@@ -0,0 +1,245 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opencensus.proto.error.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option java_multiple_files = true;
+option java_package = "io.opencensus.proto.error.v1";
+option java_outer_classname = "ErrorProto";
+
+option go_package = "github.com/census-instrumentation/opencensus-proto/gen-go/error/v1";
+
+// Error event that is generated by integration or application.
+message Error {
+  // Time when the error occurred.
+  // This field is required.  
+  google.protobuf.Timestamp time = 1;
+
+  // Error description such as exception message.
+  // This field is required.  
+  string message = 2;
+
+  // TraceId of the span in which scope the error occurred. 
+  // Must be set if there is a current span or it was explicitly
+  // provided by the caller
+  bytes trace_id = 3;
+
+  // SpanId of the span in which scope the error occurred. 
+  // Must be set if there is a current span or it was explicitly
+  // provided by the caller
+  bytes span_id = 4;
+
+  // Describes additional error details. This data should be 
+  // provided by integration site or application when error is 
+  // reported
+  message ErrorDetails
+  {
+    // Type of the error such as Exception/Error class type
+    string error_type = 1;  
+
+    // Parsed stack trace that contains line number, file info, method info, etc...
+    // Agent exporter must to it's best to parse stack trace
+    // unless agent instructed exporter to send raw format only.
+    StackTrace parsed_stack_trace = 2;
+
+    // Describes stack trace in the format convenient for specific language and exporter
+    oneof raw_stack_trace
+    {
+      // Raw string value returned by: 
+      // Java: Must be the return value of Throwable.printStackTrace().
+      // Python: Must be the return value of traceback.format_exc().
+      // JavaScript: Must be the value of error.stack as returned by V8.
+      // Ruby: Must contain joined frames returned by Exception.backtrace.
+      // C#: Must be the return value of Exception.ToString().
```

Comment:
Perhaps `new StackTrace(Exception).ToString()`? ToString will include message.

Also, in C# there are two stacks - unwind stack which exception was thrown thru and current thread stack. Perhaps those should be merged